tests/modules/encoder_base_test.py

```diff
@@ -0,0 +1,161 @@
+# pylint: disable=no-self-use,protected-access
+import numpy
+import torch
+from torch.autograd import Variable
+from torch.nn import LSTM
+
+from allennlp.modules.encoder_base import _EncoderBase
+from allennlp.common.testing import AllenNlpTestCase
+from allennlp.nn.util import sort_batch_by_length, get_lengths_from_binary_sequence_mask
+
+
+class TestEncoderBase(AllenNlpTestCase):
+
+    def setUp(self):
+        super(TestEncoderBase, self).setUp()
+        self.lstm = LSTM(bidirectional=True, num_layers=3, input_size=3, hidden_size=7, batch_first=True)
+        self.encoder_base = _EncoderBase(stateful=True)
+
+        tensor = Variable(torch.rand([5, 7, 3]))
+        tensor[1, 6:, :] = 0
+        tensor[2, :, :] = 0  # <= completely masked
+        tensor[3, 2:, :] = 0
+        tensor[4, :, :] = 0  # <= completely masked
+        self.tensor = tensor
```

Comment:
Do you need to zero out the masked values in the tensor?  I thought you just needed to set the mask to zeros.

Reply:
No, you shouldn't need to zero out the tensor.